apps/dashboard/src/app/onboarding/getting-started/getting-started.component.ts

```diff
@@ -0,0 +1,237 @@
+                  this.itemsToUpload = itemsToUpload;
+                  this.cdr.markForCheck();
+                  return itemsToUpload;
+                }),
+              )
+            : of(linkItems);
+        }),
+        switchMap((uploadedItems) => {
+          return forkJoin([this.sdk.currentAccount.pipe(take(1)), this.sdk.currentKb.pipe(take(1))]).pipe(
+            switchMap(([account, kb]) =>
+              this.uploadService.getResourceStatusCount().pipe(
+                // repeat until allProcessed but with a step-back increasing by 3s at every step but maxing out at POLLING_DELAY
+                repeat({ delay: (count) => timer(Math.min(POLLING_DELAY, count * 3000)) }),
+                takeUntil(this.allProcessed),
+                switchMap((results) =>
+                  this.sdk.nuclia.db
+                    .getProcessingStatus(account.id)
+                    .pipe(map((processingStatus) => ({ processingStatus, results }))),
+                ),
+                tap(({ processingStatus, results }) => {
+                  const resources = results.resources || {};
+                  const resourceList = Object.values(resources);
+                  this.itemsToUpload = uploadedItems.map((item) => {
+                    if (item.uploadFailed) {
+                      return item;
+                    }
+                    const iResource = item.uuid
+                      ? resources[item.uuid]
+                      : resourceList.find((resource) => resource.title === item.title);
+                    if (iResource) {
+                      const resource = new Resource(this.sdk.nuclia, kb.id, iResource);
+                      const placeInQueue = this.uploadService.getResourcePlaceInProcessingQueue(
+                        resource,
+                        processingStatus,
+                      );
+                      item.processed =
+                        resource.metadata?.status === RESOURCE_STATUS.PROCESSED ||
+                        (placeInQueue !== null && placeInQueue < 0);
+                      if (!item.processed) {
+                        item.processing = true;
+                        item.estimation =
+                          placeInQueue === null
+                            ? AVERAGE_PROCESSING_TIME
+                            : Math.max(placeInQueue * AVERAGE_PROCESSING_TIME, 0);
+                      }
+                    }
+                    return item;
+                  });
+
+                  // if everything is processed or failed, we stop polling processing status and enable next button
+                  if (this.itemsToUpload.every((item) => item.processed || item.uploadFailed)) {
+                    this.allProcessed.next();
+                    this.nextDisabled = false;
+                    this.totalEstimatedTime = 0;
+                  } else {
+                    this.totalEstimatedTime = this.itemsToUpload.reduce((total, item) => {
```

Comment:
would it be more accurate to calculate it by getting the item with maximum processing time, instead of adding all the times ?

Reply:
I'm not sure what you mean by "to calculate it by getting the item with maximum processing time". But in any case, the current naive approach is temporary, the time for the backend to implement a new endpoint providing us better estimations based on uploaded resources type and size.